cpp/rss_subset_aggregator.cc

```diff
@@ -0,0 +1,311 @@
+
+    HarvestedRSSItem(const SyndicationFormat::Item item, const std::string feed_title, const std::string feed_url)
+        : item_(item), feed_title_(feed_title), feed_url_(feed_url) {}
+};
+
+
+struct ChannelDesc {
+    std::string title_;
+    std::string link_;
+public:
+    ChannelDesc(const std::string &title, const std::string &link)
+        : title_(title), link_(link) { }
+};
+
+
+const std::map<std::string, ChannelDesc> subsystem_type_to_channel_desc_map = {
+    { "relbib",  ChannelDesc("RelBib RSS Aggregator",  "https://relbib.de/")                },
+    { "ixtheo",  ChannelDesc("IxTheo RSS Aggregator",  "https://itheo.de/")                 },
+    { "krimdok", ChannelDesc("KrimDok RSS Aggregator", "https://krimdok.uni-tuebingen.de/") },
+};
+
+
+std::string GetChannelDescEntry(const std::string &subsystem_type, const std::string &entry) {
+    const auto subsystem_type_and_channel_desc(subsystem_type_to_channel_desc_map.find(subsystem_type));
+    if (subsystem_type_and_channel_desc == subsystem_type_to_channel_desc_map.cend())
+        LOG_ERROR("unknown subsystem type \"" + subsystem_type + "\"!");
+
+    if (entry == "title")
+        return subsystem_type_and_channel_desc->second.title_;
+    if (entry == "link")
+        return subsystem_type_and_channel_desc->second.link_;
+    LOG_ERROR("unknown entry name \"" + entry + "\"!");
+}
+
+
+void WriteRSSFeedXMLOutput(const std::string &subsystem_type, const std::vector<HarvestedRSSItem> &harvested_items,
+                           XmlWriter * const xml_writer)
+{
+    xml_writer->openTag("rss", { { "version", "2.0" }, { "xmlns:tuefind", "https://github.com/ubtue/tuefind" } });
+    xml_writer->openTag("channel");
+    xml_writer->writeTagsWithData("title", GetChannelDescEntry(subsystem_type, "title"));
+    xml_writer->writeTagsWithData("link", GetChannelDescEntry(subsystem_type, "link"));
+    xml_writer->writeTagsWithData("description", "RSS Aggregator");
+
+    for (const auto &harvested_item : harvested_items) {
+        xml_writer->openTag("item");
+
+        const auto title(harvested_item.item_.getTitle());
+        if (not title.empty())
+            xml_writer->writeTagsWithData("title", harvested_item.item_.getTitle());
+
+        xml_writer->writeTagsWithData("link", harvested_item.item_.getLink());
+
+        const auto description(HtmlUtil::ShortenText(harvested_item.item_.getDescription(), /*max_length = */500));
+        if (not description.empty())
+            xml_writer->writeTagsWithData("description", description);
+
+        xml_writer->writeTagsWithData("pubDate",
+                                      TimeUtil::TimeTToString(harvested_item.item_.getPubDate(), TimeUtil::RFC822_FORMAT,
+                                                                     TimeUtil::UTC));
+        xml_writer->writeTagsWithData("guid", harvested_item.item_.getId());
+        xml_writer->writeTagsWithData("tuefind:rss_title", harvested_item.feed_title_);
```

Comment:
The entire `xmlns:tuefind` namespace and all related entries may be removed, since TueFind will be able to get the related values from the db table in the future, see ubtue/tuefind#1412

Reply:
Then what would be a good strategy here?

Reply:
I have eliminated any references to the tuefind namespace.